core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1949,20 +2038,35 @@ class SparkContext(config: SparkConf) extends Logging {
 
   /**
    * Run a job on all partitions in an RDD and return the results in an array.
+   *
+   * @param rdd target RDD to run tasks on
+   * @param func a function to run on each partition of the RDD
+   * @return in-memory collection with a result of the job (each collection element will contain
+   * a result from one partition)
    */
   def runJob[T, U: ClassTag](rdd: RDD[T], func: (TaskContext, Iterator[T]) => U): Array[U] = {
     runJob(rdd, func, 0 until rdd.partitions.length)
   }
 
   /**
-   * Run a job on all partitions in an RDD and return the results in an array.
+   * Run a job on all partitions in an RDD and return the results in an array, but take a function
+   * of type `Iterator[T] => U` instead of `(TaskContext, Iterator[T]) => U`.
```

Comment:
Two `runJob` methods had exactly the same javadoc, I added this line (similar to the already existing line 2020) to show the difference between them.